cmake/catkin_libraries.cmake

```diff
@@ -81,14 +82,15 @@ macro(catkin_pack_libraries_with_build_configuration VAR)
         message(FATAL_ERROR "catkin_pack_libraries_with_build_configuration() the list of libraries '${_argn}' ends with '${lib}' which is a build configuration keyword and must be followed by a library")
       endif()
       list(GET _argn ${_index} library)
-      list(APPEND ${VAR} "${lib}${CATKIN_BUILD_CONFIGURATION_KEYWORD_SEPARATOR}${library}")
+      list(APPEND result "${lib}${CATKIN_BUILD_CONFIGURATION_KEYWORD_SEPARATOR}${library}")
     else()
-      list(APPEND ${VAR} "${lib}")
+      list(APPEND result "${lib}")
     endif()
     math(EXPR _index "${_index} + 1")
   endwhile()
+  set(${VAR} "${result}" PARENT_SCOPE)
   #debug_message(10 "catkin_pack_libraries_with_build_configuration(${VAR} ${_argn}) ${${VAR}}")
```

Comment:
Just to be safe if someone enables this line in the future can you also update it to use `${result}` instead of `${${VAR}}`.